src/System/Validation/ValidatedNotNullAttribute.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.System.Validation
+{
+	/// <summary>
+	/// Indicates to Code analysis tools that a method validates a particular parameter to not be null

```

Comment:
There is some useful information in your PR description about the cases where this is beneficial. I'd encourage you to add those details into this comment so that they aren't lost over time.

Reply:
I will look into this!

Reply:
Thanks for adding the details!